plugins/search-backend/src/actions/createQueryAction.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ActionsRegistryService } from '@backstage/backend-plugin-api/alpha';
+import { SearchEngine } from '@backstage/plugin-search-backend-node';
+import { SearchIndexService } from '@backstage/plugin-search-backend-node/alpha';
+
+export const createQueryAction = ({
+  engine,
+  searchIndexService,
+  actionsRegistry,
+}: {
+  engine: SearchEngine;
+  searchIndexService: SearchIndexService;
+  actionsRegistry: ActionsRegistryService;
+}) => {
+  const [firstType, ...otherTypes] = Object.keys(
+    searchIndexService.getDocumentTypes(),
```

Comment:
This might actually not work as the searchIndexService might not be initialized 🤔